src/Tizen.Network.Nsd/Tizen.Network.Nsd/DnssdService.cs

```diff
@@ -217,15 +219,77 @@ public IPAddressInformation IP
                 return IPAddressInstance;
             }
         }
+        
+        /// <summary>
+        /// Gets all TXT records in the form of key and value.
+        /// </summary>
+        /// <remarks>
+        /// After adding the TXT record call for get TXT record.
+        /// </remarks>
+        /// <since_tizen> 9 </since_tizen>
+        /// <param name="map">gets the each record key, value in the map</param>
+        /// <feature>http://tizen.org/feature/network.service_discovery.dnssd</feature>
+        /// <exception cref="NotSupportedException">Thrown when DNS-SD is not supported.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when any other error occurred.</exception>
+        public void GetTxtRecord(Dictionary<string, string> map)
+        {
+            IntPtr data;
+            ushort length;
 
-        private void GetTxtRecord(out ushort length, out byte[] value)
+            int ret = Interop.Nsd.Dnssd.GetAllTxtRecord(_serviceHandle, out length, out data);
+            if (ret != (int)DnssdError.None)
+            {
+                Log.Error(Globals.LogTag, "Failed to get the TXT record, Error: " + (DnssdError)ret);
+                NsdErrorFactory.ThrowDnssdException(ret);
+            }
+            if (length > 0)
+            {
+                byte[] managedArray = new byte[length];
+                Marshal.Copy(data, managedArray, 0, length);
+                var index = 0;
+                while (index < length)
+                {
+                    var txtLen = managedArray[index++];
+                    if (txtLen > length)
+                        Log.Debug(Globals.LogTag, "Invalid data");
+                    
+                    string record = Encoding.UTF8.GetString(managedArray, index, txtLen);
+                    string[] list = record.Split('=');
+                    map.Add(list[0], list[1]);
+                    index += txtLen;
+                }
+            }
+        }
+        
+        /// <summary>
+        /// Gets all raw TXT records.
+        /// </summary>
+        /// <remarks>
+        /// After adding the TXT record call for get TXT record.
+        /// </remarks>
+        /// <since_tizen> 4 </since_tizen>
+        /// <param name="length">gets the length of total txt recods. It txt record not available length value will be zero</param>
+        /// <param name="value">The value of the TXT record. If txt record not availabe added with an empty value.</param>
+        /// <feature>http://tizen.org/feature/network.service_discovery.dnssd</feature>
+        /// <exception cref="NotSupportedException">Thrown when DNS-SD is not supported.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when any other error occurred.</exception>
+        public void GetTxtRecord(out ushort length, out byte[] value)
```

Comment:
Is it possible just return a string type or byte[]? and length is not necessary because users can use Length property of string or byte[].

Reply:
Length parameter  is required in this api to know the total length of raw txt stored in byte array.

Reply:
@prasadam-p look at this
array has Length property
https://docs.microsoft.com/en-us/dotnet/api/system.array.length?view=net-5.0

Reply:
applied review comment